otherlibs/runtime_events/runtime_events_consumer.c

```diff
@@ -216,21 +217,29 @@ caml_runtime_events_create_cursor(const char_os* runtime_events_path, int pid,
 
   /* This cast is necessary for compatibility with Illumos' non-POSIX
     mmap/munmap */
-  cursor->metadata = (struct runtime_events_metadata_header *)
-                      mmap(NULL, cursor->ring_file_size_bytes, mmap_prot,
+  cursor->map = (void*) mmap(NULL, cursor->ring_file_size_bytes, mmap_prot,
                           MAP_SHARED, ring_fd, 0);
 
-  if( cursor->metadata == MAP_FAILED ) {
+  if( cursor->map == MAP_FAILED ) {
     caml_stat_free(cursor);
     caml_stat_free(runtime_events_loc);
     return E_MAP_FAILURE;
   }
 #endif
 
+  cursor->metadata = *(struct runtime_events_metadata_header*)cursor->map;
```

Comment:
```suggestion
  memcpy(cursor->metadata, cursor->map,
         sizeof(struct runtime_events_metadata_header));
```

Is this line clearer with a `memcpy`? Could it avoid problems with pointer alignment?